sw/device/silicon_creator/lib/drivers/retention_sram.c

```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/retention_sram.h"
+
+#include <assert.h>
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+volatile retention_sram_t *retention_sram_get(void) {
+  static_assert(sizeof(retention_sram_t) == TOP_EARLGREY_RAM_RET_AON_SIZE_BYTES,
+                "Unexpected retention SRAM size.");
+  return (volatile retention_sram_t *)TOP_EARLGREY_RAM_RET_AON_BASE_ADDR;
+}
+
+void retention_sram_clear(void) {
+  *((volatile retention_sram_t *)TOP_EARLGREY_RAM_RET_AON_BASE_ADDR) =
```

Comment:
If I recall correctly Chris didn't think (or wasn't sure if) we could rotate the scrambling key because it would be cleared back to 0 when the device reset thereby effectively wiping the retention SRAM. I can give it a go though and see if that is indeed the behavior.

Reply:
Ah, found the relevant discussion: https://github.com/lowRISC/opentitan/pull/7220#issuecomment-873117531

Reply:
nice thread :) We should move it to an issue so that we can handle it as an open. If we leave retention ram unscrambled, we should add a warning to the c struct, and also add a checksum or digest to protect the integrity as an attacker may be able to manipulate the data to abuse boot services. Or perhaps the multibit encoding of bot_services is sufficient.

Reply:
Done (https://github.com/lowRISC/opentitan/issues/7895)

Reply:
Done